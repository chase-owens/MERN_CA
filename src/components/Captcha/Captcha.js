import React from 'react';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import Button from '@material-ui/core/Button';
import { theme } from '../../styles/theme';
import { isMobile } from 'react-device-detect';
import { withStyles } from '@material-ui/core/styles';

import withWidth from '@material-ui/core/withWidth';
import { unstable_useMediaQuery as useMediaQuery } from '@material-ui/core/useMediaQuery';

const boxes = [0, 1, 2, 3, 4, 5, 6, 7, 8];
const logoPath = require('images/Logo.png');

const styles = theme => ({
  box: {
    '&:hover': {
      cursor: 'pointer'
    }
  }
});

const Captcha = ({ selected, handleChange, classes }) => {
  return (
    <div style={{ width: '100%' }}>
      {useMediaQuery(theme.breakpoints.down(675)) && (
        <Card
          style={{
            width: isMobile ? '90vw' : '30vw',
            background: theme.palette.primary.main,
            display: 'block',
            margin: 'auto'
          }}
        >
          <Grid
            spacing={8}
            justify='space-around'
            container
            style={{
              height: isMobile ? '90vw' : '30vw',
              width: '100%',
              margin: 'auto',
              background: theme.palette.ternary.main,
              overflow: 'hidden',
              padding: 4
            }}
          >
            {boxes.map((box, i) => (
              <Grid
                className={classes.box}
                key={box}
                value={i}
                item
                onClick={e => handleChange(i)}
                style={{
                  height: '33%',
                  width: '33%',
                  borderRadius: '5px',
                  border:
                    i === 4
                      ? selected.includes(4)
                        ? `4px solid ${theme.palette.primary.dark}`
                        : `2px solid ${theme.palette.ternary.main}`
                      : `2px solid ${theme.palette.ternary.main}`,
                  background:
                    i !== 4
                      ? selected.includes(i)
                        ? theme.palette.secondary.dark
                        : theme.palette.secondary.light
                      : theme.palette.primary.main
                }}
              >
                {i === 4 && (
                  <img
                    src={logoPath}
                    alt=''
                    style={{
                      height: isMobile ? '25vw' : '8vw',
                      marginTop: isMobile
                        ? selected.includes(4)
                          ? 7
                          : 8
                        : selected.includes(4)
                        ? -3.8
                        : -2,
                      marginLeft: isMobile
                        ? selected.includes(4)
                          ? 12
                          : 13
                        : selected.includes(4)
                        ? -0.2
                        : 2
                    }}
                  />
                )}
              </Grid>
            ))}
          </Grid>
        </Card>
      )}
      {useMediaQuery(theme.breakpoints.up(675)) && (
        <Card
          style={{
            width: '30vw',
            background: theme.palette.primary.main
          }}
        >
          <Grid
            spacing={8}
            justify='space-around'
            container
            style={{
              height: '30vw',
              width: '100%',
              margin: 'auto',
              background: theme.palette.ternary.main,
              overflow: 'hidden',
              padding: 4
            }}
          >
            {boxes.map((box, i) => (
              <Grid
                className={classes.box}
                key={box}
                value={i}
                item
                onClick={e => handleChange(i)}
                style={{
                  height: '33%',
                  width: '33%',
                  borderRadius: '5px',
                  border:
                    i === 4
                      ? selected.includes(4)
                        ? `8px solid ${theme.palette.primary.dark}`
                        : `4px solid ${theme.palette.ternary.main}`
                      : `4px solid ${theme.palette.ternary.main}`,
                  background:
                    i !== 4
                      ? selected.includes(i)
                        ? theme.palette.secondary.dark
                        : theme.palette.secondary.light
                      : theme.palette.primary.main
                }}
              >
                {i === 4 && (
                  <img
                    src={logoPath}
                    alt=''
                    style={{
                      height: '8vw',
                      marginTop: selected.includes(4) ? 0 : '0.2vw',
                      marginLeft: selected.includes(4) ? '0.65vw' : '0.9vw'
                    }}
                  />
                )}
              </Grid>
            ))}
          </Grid>
        </Card>
      )}
    </div>
  );
};

export default withStyles(styles)(withWidth({ withTheme: true })(Captcha));
