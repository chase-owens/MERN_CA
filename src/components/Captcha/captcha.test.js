import React from 'react';
import Captcha from './Captcha';
import { shallow } from 'enzyme';

describe('Captcha', () => {
  it('renders without crashing', () => {
    const captcha = shallow(<Captcha />);
    expect(captcha).toMatchSnapshot();
  });
});
