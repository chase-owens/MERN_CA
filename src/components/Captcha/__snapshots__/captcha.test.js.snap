// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Captcha renders without crashing 1`] = `
<WithTheme(WithWidth(Captcha))
  classes={
    Object {
      "box": "WithTheme-WithWidth-Captcha---box-1",
    }
  }
/>
`;
